packages/kn-plugin-workflow/pkg/common/checks.go

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package common
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+)
+
+func CheckJavaDependencies() error {
+	fmt.Println("✅ Checking dependencies...")
+	if err := checkJava(); err != nil {
+		return fmt.Errorf("%w", err)
+	}
+	if err := checkMaven(); err != nil {
+		return fmt.Errorf("%w", err)
+	}
+	return nil
+}
+
+func checkJava() error {
+	javaCheck := exec.Command("java", "-version")
+	version, err := javaCheck.CombinedOutput()
+	if err != nil {
+		fmt.Println("ERROR: Java not found")
+		fmt.Println("At least Java 11 is required to use this command")
+		return err
+	}
+	userJavaVersion, err := parseJavaVersion(string(version))
+	if err != nil {
+		return fmt.Errorf("error while parsing Java version: %w", err)
+	}
+
+	if userJavaVersion < JAVA_VERSION {
+		fmt.Printf("ERROR: Please make sure you are using Java version %.2d or later", JAVA_VERSION)
+		fmt.Println("Installation stopped. Please upgrade Java and run again")
+		os.Exit(1)
+	} else {
+		fmt.Println(" - Java version check.")
+	}
+	return nil
+}
+
+func checkMaven() error {
+	mavenCheck := exec.Command("mvn", "--version")
+	version, err := mavenCheck.CombinedOutput()
+	if err != nil {
+		fmt.Println("ERROR: Maven not found")
+		fmt.Println("At least Maven 3.8.1 is required to use this command")
```

Comment:
```suggestion
		fmt.Println("At least Maven 3.8.1 is required to use this command")
```

Same thing here. So we can change it in one place and propagate it to the rest of the code base.